internal/gosourcefile/gosourcefile.go

```diff
@@ -30,10 +32,22 @@ func (f *GoSourceFile) Incubate(virus viruses.Virus) []*goinfectedfile.GoInfecte
 		panic(fmt.Errorf("failed parsing file '%s': %w", f.relativePath, err))
 	}
 
+	cfg := types.Config{
+		Importer: importer.Default(),
+	}
+
+	info := types.Info{
+		Defs:  map[*ast.Ident]types.Object{},
+		Uses:  map[*ast.Ident]types.Object{},
+		Types: map[ast.Expr]types.TypeAndValue{},
+	}
+
+	_, _ = cfg.Check(f.relativePath, fileSet, []*ast.File{fileTree}, &info)
```

Comment:
Error should be checked instead of silently discarded, but I did not want to make too many changes at once. Please advise.

Reply:
All good. At this point, given there's no point in continuing as there's no way to recover from this particular error, I'd just panic like the parsing just a few lines above: https://github.com/gtramontina/ooze/blob/40b4806bd7c585c94d1d4270a3c9bd54e125661c/internal/gosourcefile/gosourcefile.go#L30.

Reply:
I see. I guess panicking should be okay here since we're running in a testing environment after all.